nav2_behavior_tree/plugins/action/planner_selector_node.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2018 Intel Corporation
+// Copyright (c) 2020 Pablo Iñigo Blasco
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <memory>
+
+#include "std_msgs/msg/string.hpp"
+
+#include "nav2_behavior_tree/plugins/action/planner_selector_node.hpp"
+
+#include "rclcpp/rclcpp.hpp"
+
+namespace nav2_behavior_tree
+{
+
+using std::placeholders::_1;
+
+PlannerSelector::PlannerSelector(
+  const std::string & name,
+  const BT::NodeConfiguration & conf)
+: BT::SyncActionNode(name, conf)
+{
+  node_ = config().blackboard->get<rclcpp::Node::SharedPtr>("node");
+
+  getInput("topic_name", topic_name_);
+
+  planner_selector_sub_ = node_->create_subscription<std_msgs::msg::String>(
+    topic_name_, 1, std::bind(&PlannerSelector::callback_planner_select, this, _1));
+}
+
+BT::NodeStatus PlannerSelector::tick()
+{
+  rclcpp::spin_some(node_);
+
+  if (last_selected_planner_.empty()) {
+    getInput("default_planner", last_selected_planner_);
+  }
+
+  setOutput("selected_planner", last_selected_planner_);
+
+  return BT::NodeStatus::SUCCESS;
+}
+
+void
+PlannerSelector::callback_planner_select(const std_msgs::msg::String::SharedPtr msg)
```

Comment:
Function naming styling should be `callbackPlannerSelect`

Reply:
This should be already fixed